display_list/effects/dl_color_source.cc

```diff
@@ -38,9 +45,26 @@ std::shared_ptr<DlColorSource> DlColorSource::MakeLinear(
     const float* stops,
     DlTileMode tile_mode,
     const DlMatrix* matrix) {
-  size_t needed = sizeof(DlLinearGradientColorSource) +
-                  (stop_count * (sizeof(DlColor) + sizeof(float)));
+  size_t needed = CalculateLinearGradientSize(stop_count);
+  void* storage = ::operator new(needed);
 
+  std::shared_ptr<DlLinearGradientColorSource> ret;
+  ret.reset(new (storage)
+                DlLinearGradientColorSource(start_point, end_point, stop_count,
+                                            colors, stops, tile_mode, matrix),
+            DlGradientDeleter);
+  return ret;
+}
+
+std::shared_ptr<DlColorSource> DlColorSource::MakeLinear(
+    const DlPoint start_point,
+    const DlPoint end_point,
+    uint32_t stop_count,
+    const DlScalar* colors_argb,
```

Comment:
Below this is referred to as just "colors"???

Reply:
thanks, done